internal/pkg/eventhandler/eventhandler.go

```diff
@@ -111,22 +111,21 @@ func (h *fnKeyHandler) KeyUp() handlerFunc {
 
 		h.currentInputSource = inputsource.Current()
 
-		if !h.doublePressed {
-			logger.Info("globe key pressed")
-
-			h.setInputSource(h.primaryInputSources[h.getNextInputSource(&h.primaryInputSources)])
-		} else {
+		if h.doublePressed && len(h.additionalInputSources) > 0 {
```

Comment:
we don't need the `len(h.additionalInputSources) > 0` condition because `h.doublePressed` will never be `true` anyway since line 94 will never allow it to

so I'm pretty confident that we can return lines 114-128 to the original state